File: PaymentService.py
import payment_gateway_sdk

class PaymentService:
    def __init__(self, secret_key):
        self.sdk = payment_gateway_sdk.SDK(secret_key)

    def create_payment_intent(self, amount_in_cents, currency):
        intent = self.sdk.create_payment_intent(amount_in_cents, currency)
        return intent['client_secret'] # Or a unique ID for client-side use

    def confirm_payment(self, payment_intent_id):
        return self.sdk.confirm_payment(payment_intent_id)

File: PaymentService.java
import com.example.paymentgateway.PaymentGatewaySDK;
import com.example.paymentgateway.models.PaymentIntent;
import com.example.paymentgateway.exceptions.PaymentException;

public class PaymentService {

    private final PaymentGatewaySDK sdk;

    public PaymentService(String secretKey) {
        this.sdk = new PaymentGatewaySDK(secretKey);
    }

    public String createPaymentIntent(long amountInCents, String currency) throws PaymentException {
        PaymentIntent intent = sdk.createPaymentIntent(amountInCents, currency);
        return intent.getClientSecret(); // Or a unique ID for client-side use
    }

    public boolean confirmPayment(String paymentIntentId) throws PaymentException {
        return sdk.confirmPayment(paymentIntentId);
    }
}
